lib/checkbool.cpp

```diff
@@ -93,27 +93,18 @@ void CheckBool::checkBitwiseOnBoolean()
     const SymbolDatabase *symbolDatabase = mTokenizer->getSymbolDatabase();
     for (const Scope * scope : symbolDatabase->functionScopes) {
         for (const Token* tok = scope->bodyStart->next(); tok != scope->bodyEnd; tok = tok->next()) {
-            if (Token::Match(tok, "(|.|return|&&|%oror%|throw|, %var% [&|]")) {
-                const Variable *var = tok->next()->variable();
-                if (isBool(var)) {
-                    bitwiseOnBooleanError(tok->next(), var->name(), tok->strAt(2) == "&" ? "&&" : "||");
-                    tok = tok->tokAt(2);
-                }
-            } else if (Token::Match(tok, "[&|] %var% )|.|return|&&|%oror%|throw|,") && (!tok->previous() || !tok->previous()->isExtendedOp() || tok->strAt(-1) == ")" || tok->strAt(-1) == "]")) {
-                const Variable *var = tok->next()->variable();
-                if (isBool(var)) {
-                    bitwiseOnBooleanError(tok->next(), var->name(), tok->str() == "&" ? "&&" : "||");
-                    tok = tok->tokAt(2);
-                }
+            if (tok->isBinaryOp() && (tok->str() == "&" || tok->str() == "|")
+                    && (astIsBool(tok->astOperand1()) || astIsBool(tok->astOperand2()))) {
+                bitwiseOnBooleanError(tok, tok->str() == "&" ? "&&" : "||");
             }
         }
     }
 }
 
-void CheckBool::bitwiseOnBooleanError(const Token *tok, const std::string &varname, const std::string &op)
+void CheckBool::bitwiseOnBooleanError(const Token *tok, const std::string &op)
 {
     reportError(tok, Severity::style, "bitwiseOnBoolean",
-                "Boolean variable '" + varname + "' is used in bitwise operation. Did you mean '" + op + "'?",
+                "Boolean is used in bitwise operation. Did you mean '" + op + "'?",
```

Comment:
Can we pass `tok->expressionString()` for `varname` so we can still get a warning like `Boolean 'a<0' is used in bitwise operation`?

Reply:
I'd be happy to. More improvement could be made, but the types of improvements I'm thinking of are beyond the scope of the original issue. It should be revisited, for sure.